File: 백준/Silver/1463. 1로 만들기/1로 만들기.py
n=int(input())
arr=[n]
count=0
finish=False
while True:
  tmp=[]
  for i in range(len(arr)):
    # print(arr[0]%3)
    if min(arr)==1:
      finish=True
      break
    if arr[i]%2==0:
      tmp.append(arr[i]//2)
    if arr[i]%3==0:
      tmp.append(arr[i]//3)
      # print(arr[i],arr[i]%3,arr[i]//3,tmp)
    tmp.append(arr[i]-1)
  if finish:
    break
  count+=1
  arr=list(set(tmp))

print(count)

File: 백준/Silver/1463. 1로 만들기/1로 만들기.js
const fs = require("fs");
let n = fs.readFileSync(0).toString().trim().split('\n');
let arr=[n]
let count=0
let finish=false
while(true){
    let tmp=[]
    for(let i=0;i<arr.length;i++){
        if(Math.min(...arr)===1){
            finish=true
            break
        }
        if(arr[i]%3===0)
            tmp.push(arr[i]/3)
        if(arr[i]%2===0)
            tmp.push(arr[i]/2)
        tmp.push(arr[i]-1)
    }
    if(finish)
        break
    arr=[...new Set(tmp)]
    count+=1
}
console.log(count)